### ISSUE TITLE
WindowsError while getting version from setup.py - caused by wrong environment

### ISSUE THREAD

**Issue by codewarrior0** [zestsoftware/zest.releaser#72]
Using python2.7 64-bit on Windows 7, with a virtualenv active.

While running `prerelease`, an odd error is raised while probing for the version number with `setup.py --version` (getting the whole traceback required a minor change to `get_setup_py_version`):

```
CRITICAL: The setup.py of this package has an error:
Traceback (most recent call last):
  File "setup.py", line 16, in <module>
    setup(
  File "c:\Users\Rio\Documents\mcedit2-distrib\mcedit2\src\mceditlib\setup.py", line 1, in <module>
    from setuptools import setup
  File "c:\Users\Rio\Documents\mcedit2-distrib\mcedit2\ENV64\lib\site-packages\setuptools\__init__.py", line 12, in <mod
ule>
    from setuptools.extension import Extension
  File "c:\Users\Rio\Documents\mcedit2-distrib\mcedit2\ENV64\lib\site-packages\setuptools\extension.py", line 8, in <mod
ule>
    from .dist import _get_unpatched
  File "c:\Users\Rio\Documents\mcedit2-distrib\mcedit2\ENV64\lib\site-packages\setuptools\dist.py", line 23, in <module>

    from setuptools.depends import Require
  File "c:\Users\Rio\Documents\mcedit2-distrib\mcedit2\ENV64\lib\site-packages\setuptools\depends.py", line 6, in <modul
e>
    from setuptools import compat
  File "c:\Users\Rio\Documents\mcedit2-distrib\mcedit2\ENV64\lib\site-packages\setuptools\compat.py", line 17, in <modul
e>
    import httplib
  File "C:\Python27-64\Lib\httplib.py", line 79, in <module>
    import mimetools
  File "C:\Python27-64\Lib\mimetools.py", line 6, in <module>
    import tempfile
  File "C:\Python27-64\Lib\tempfile.py", line 35, in <module>
    from random import Random as _Random
  File "C:\Python27-64\Lib\random.py", line 879, in <module>
    _inst = Random()
  File "C:\Python27-64\Lib\random.py", line 97, in __init__
    self.seed(x)
  File "C:\Python27-64\Lib\random.py", line 111, in seed
    a = long(_hexlify(_urandom(16)), 16)
WindowsError: [Error -2146893818] Invalid Signature

CRITICAL: No version found.
```

This error is not raised when running setup.py by itself.

After some investigating, I learned this error is the Windows DLL loader (Windows Side-By-Side) unable to locate the crypto libraries - because the SYSTEMROOT environment variable is not present! The root cause is in your `utils.system` function, which substitutes a nearly-empty env dict whenever the exe passed to `utils.system` is the Python exe used to run `prerelease`.

Unless I'm missing something, the fix is to copy `os.environ` and update its `PYTHONPATH` instead of create a new dict for env.

**Comment by reinout** [comment]
I saw a mention about `SYSTEMROOT` on the python distutils mailinglist a while ago, probably *something* changed a while a go.

Thanks for the patch, I've merged it. I'm going to merge another branch, too, and make a release today.

**Comment by reinout** [comment]
I released 3.55. Can you check if it works for you?

**Comment by codewarrior0** [comment]
Thanks for the release. Will give it a try once I get all these other fires put out.
